configure.ac

```diff
@@ -74,8 +72,14 @@ AC_CHECK_LIB(dl, dlopen, [],
 dnl Checks for libraries
 AC_CHECK_LIB(pthread, pthread_mutex_init, [],
     AC_MSG_ERROR([pthread_mutex_init() not found.  libfabric requires libpthread.]))
-AC_CHECK_LIB(rt, clock_gettime, [],
-    AC_MSG_ERROR([clock_gettime() not found.  libfabric requires librt.]))
+
+AC_SEARCH_LIBS([clock_gettime],[rt posix4],
+		[AC_DEFINE([HAVE_CLOCK_GETTIME], [1],
+			[Define to 1 if have clock_gettime.])],
+		[AS_IF([test -z $(expr "$host_os" : '.*\(darwin\)')],
+			[AC_MSG_ERROR([clock_gettime not found.])],
```

Comment:
Well see the issue is that clock_gettime is defined on BSD but not OS X. So if clock_gettime is not available in librt or posix4 then it could be OS X or some obscure system. If it's OS X we have a fix for that which relies on OS X specific headers (mach/clock.h, mach/mach.h). If it isn't in librt, posix4, or an OS X machine then we can't handle that. I don't know how to make it more general, and it's necessary to avoid trying to build on obscure systems.